### DISCUSSION TITLE
Empty substructure pattern matching behavior?

### DISCUSSION THREAD

**Discussion by drassokhin** [rdkit/rdkit#6955] [General]
The Python code below
 ```python
import re
from rdkit import Chem

print(re.findall('', 'some string'))

substructure = Chem.MolFromSmarts('')
m = Chem.MolFromSmiles('')
print(f'Empty mol:     has matches? {m.HasSubstructMatch(substructure)}')
m = Chem.MolFromSmiles('c1cc(C(=O)O)c(OC(=O)C)cc1')
print(f'Non-empty mol: has matches? {m.HasSubstructMatch(substructure)}')
matches = m.GetSubstructMatches(substructure)
print(matches)
```
, when executed in a Jupyter notebook, will produce the following output:
```
['', '', '', '', '', '', '', '', '', '', '', '']
Empty mol:     has matches? False
Non-empty mol: has matches? False
()
```
An empty regular expression matches any string. In RDKit, an empty substructure pattern does not match any molecule, including an empty molecule. Is this the expected behavior?

**Comment by bp-kelley** [comment]
The result of a subtruct match is the list of atom indexes found in the match.  In this case, no atoms were found hence there is no match, this is just the definition used by GetSubstructMatches and inherited by HasSubstructMatch.  No substructure, no match.
